packages/storybook-ui-components/tsconfig.json

```diff
@@ -0,0 +1,66 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "target": "es5",
+    "lib": ["es5", "es6", "es7", "es2017", "dom"],
+    "esModuleInterop": true,
+    "sourceMap": true,
+    "allowJs": false,
+    "jsx": "react",
+    "moduleResolution": "node",
+    "rootDir": "stories",
+    "baseUrl": "stories",
+    "forceConsistentCasingInFileNames": true,
+    "noImplicitReturns": true,
+    "noImplicitThis": true,
+
+    /**
+      TODO(j-low): set noImplicitAny to `true`
+
+      Explanation: setting noImplicitAny to `true` creates a problem where
+      we import the `@cockroachlabs/ui-components` subpackage. Specifically it throws
+      the following error:
+
+      ```
+      Could not find a declaration file for module '@cockroachlabs/ui-components'
+      ...implicitly has an 'any' type.
+
+      Try `npm install @types/cockroachlabs__ui-components` if it exists or add
+      a new declaration (.d.ts) file containing `declare module '@cockroachlabs/ui-components';`
+      ```
+
+      To solve this is issue seems to require reconfiguring Webpack and Typescript
```

Comment:
After talking to someone who works on typescript library, I don't think we need for the types to be in one file. The second option in https://medium.com/@NiGhTTraX/how-to-set-up-a-typescript-monorepo-with-lerna-c6acda7d4559 (using `composite: true`) is what they do for their company's shared library. Unfortunately that's not open sourced but I think the article provides a good overview.